File: app/src/main/java/com/hanix/randomgame/view/event/BackPressCloseHandler.java
package com.hanix.randomgame.view.event;

import android.app.Activity;
import android.webkit.WebView;

import com.hanix.randomgame.R;

public class BackPressCloseHandler {

    private long backKeyPressedTime = 0;

    private Activity activity;
    private WebView mWebView;

    public BackPressCloseHandler(Activity context, WebView webView)
    {
        this.activity = context;
        mWebView = webView;
    }

    public void onBackPressed()
    {
        if (System.currentTimeMillis() > backKeyPressedTime + 2000) {
            backKeyPressedTime = System.currentTimeMillis();
            if(mWebView != null && mWebView.canGoBack()) {
                mWebView.goBack();
            }else {
                android.os.Process.killProcess(android.os.Process.myPid());
                System.exit(0);
            }
            return;
        }
        if (System.currentTimeMillis() <= backKeyPressedTime + 2000) {
            activity.finish();
            activity.overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_right);
        }
    }

}

File: app/src/main/java/com/hanix/randomgame/view/event/BackPressCloseHandler.kt
package com.hanix.randomgame.view.event

import android.app.Activity
import com.hanix.randomgame.R
import com.hanix.randomgame.common.utils.ToastUtil.setToastLayout

/**
 * 뒤로가기(물리키) 터치 시 이벤트 처리
 */
class BackPressCloseHandler(private val activity: Activity) {

    private var backKeyPressedTime: Long = 0

    /**
     * 단순하게 activity 를 끄기만 할 때 사용함
     */
    fun onBackPressedInActivity() {
        activity.finish()
        activity.overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_right)
    }

    /**
     * mainActivity 등 주요 화면에서 물리키 작동 시
     */
    private fun onBackPress() {
        if (System.currentTimeMillis() > backKeyPressedTime + 2000) {
            backKeyPressedTime = System.currentTimeMillis()
            setToastLayout(activity, "'뒤로' 버튼을 한 번 더 누르시면 종료됩니다.", activity.applicationContext)
            return
        }

        if (System.currentTimeMillis() <= backKeyPressedTime + 2000) activity.finish()
    }
}